android/engine/src/main/java/org/smartregister/fhircore/engine/ui/login/LoginViewModel.kt

```diff
@@ -91,11 +111,41 @@ constructor(
       }
     }
 
-  private fun storeUserPreferences(responseBody: ResponseBody) {
-    val responseBodyString = responseBody.string()
-    Timber.d(responseBodyString)
-    val userResponse = responseBodyString.decodeJson<UserInfo>()
-    sharedPreferences.write(USER_INFO_SHARED_PREFERENCE_KEY, userResponse.encodeJson())
+  private fun storeUserPreferences(userInfo: UserInfo) {
+    sharedPreferences.write(USER_INFO_SHARED_PREFERENCE_KEY, userInfo.encodeJson())
+  }
+
+  fun callPractitionerDetails(userResponse: UserInfo) {
+    runBlocking {
```

Comment:
make function suspend. we added this just to run test temporarily

Reply:
Currently using it because I am saving resource into ResourceEntity table using `fhirEngine.save(it)` method which is a suspend function and if make it suspend then I need to make other functions calling this method and there's chain of methods and in which there's a retrofit call as well `handleResponse()` which can be a suspend method.